websocket_push_session.go

```diff
@@ -113,13 +117,30 @@ func (s *wsPushSession) String() string {
 // SetClaims implements elemental.ClaimsHolder.
 func (s *wsPushSession) SetClaims(claims []string) {
 
+	s.claimsLock.Lock()
+	defer s.claimsLock.Unlock()
+
 	s.claims = claims
 	s.claimsMap = claimsToMap(claims)
 }
 
+func (s *wsPushSession) Claims() []string {
+
+	s.claimsLock.RLock()
+	defer s.claimsLock.RUnlock()
+
+	return s.claims
+}
+
+func (s *wsPushSession) ClaimsMap() map[string]string {
+
+	s.claimsLock.RLock()
+	defer s.claimsLock.RUnlock()
+
+	return s.claimsMap
```

Comment:
Return a copy (map)

Reply:
Isn't a copy still a read of a map? If so, we will run into a data race when `SetClaims` is called without the lock.